src/oas2/transformers/responses.ts

```diff
@@ -1,51 +1,72 @@
+import { isResponseObject } from '../guards';
 import { getExamplesFromSchema } from './getExamplesFromSchema';
 import { translateToHeaderParams } from './params';
 
-function translateToResponse(produces: string[], response: Response, statusCode: string): IHttpOperationResponse {
-  const headers = translateToHeaderParams(response.headers || {});
-  const objectifiedExamples = chain(
-    response.examples || (response.schema ? getExamplesFromSchema(response.schema) : void 0),
-  )
-    .mapValues((value, key) => ({ key, value }))
-    .values()
-    .value();
-
-  const contents = produces.map(produceElement => ({
-    mediaType: produceElement,
-    schema: response.schema as JSONSchema4,
-    examples: objectifiedExamples.filter(example => example.key === produceElement),
-  }));
-
-  const translatedResponses = {
-    code: statusCode,
-    description: response.description,
-    headers,
-    contents,
-  };
+function responseTransformer(document: DeepPartial<Spec>) {
+  return function translateToResponse(
+    produces: string[],
+    response: unknown,
+    statusCode: string,
+  ): Optional<IHttpOperationResponse> {
+    response = maybeResolveLocalRef(document, response);
+    if (!isResponseObject(response)) return;
 
-  const foreignExamples = objectifiedExamples.filter(example => !produces.includes(example.key));
-  if (foreignExamples.length > 0) {
-    if (translatedResponses.contents.length === 0)
-      translatedResponses.contents[0] = {
-        mediaType: '',
-        schema: {},
-        examples: [],
-      };
+    const headers = translateToHeaderParams(response.headers || {});
+    const objectifiedExamples = chain(
+      response.examples || (response.schema ? getExamplesFromSchema(response.schema) : void 0),
+    )
+      .mapValues((value, key) => ({ key, value }))
+      .values()
+      .value();
 
-    translatedResponses.contents[0].examples!.push(...foreignExamples);
-  }
+    const contents = produces.map(produceElement => ({
+      mediaType: produceElement,
+      schema: (response as Response).schema as JSONSchema4,
+      examples: objectifiedExamples.filter(example => example.key === produceElement),
+    }));
+
+    const translatedResponses = {
+      code: statusCode,
+      description: response.description,
+      headers,
+      contents,
+    };
+
+    const foreignExamples = objectifiedExamples.filter(example => !produces.includes(example.key));
+    if (foreignExamples.length > 0) {
+      if (translatedResponses.contents.length === 0)
+        translatedResponses.contents[0] = {
+          mediaType: '',
+          schema: {},
+          examples: [],
+        };
 
-  return translatedResponses;
+      translatedResponses.contents[0].examples!.push(...foreignExamples);
+    }
+
+    return translatedResponses;
+  };
 }
 
 export function translateToResponses(
-  responses: { [name: string]: Response },
+  document: DeepPartial<Spec>,
+  responses: unknown,
```

Comment:
Why did you change this to `unknown`?

Reply:
Because it is unknown - we do the guard check later. It's similar to the oas3 transformer - I guess just had not been improved here yet?

Reply:
Ah yeah ok — that makes sense.